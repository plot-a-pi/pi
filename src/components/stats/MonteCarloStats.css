.MonteCarloStats{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 90vw;
  height: 20vw;
  border-radius: .5em;
  margin: 2vw;
  text-align: center;
  
  .card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-radius: 100%;
    background: rgb(197, 209, 240);
    height: 100%;
    width: 28vw;
  }

  h2 {
    font-size: 2.5vw;
    font-weight: bolder;
    height: 4vw;
    margin-bottom: 2vw;
  }

  h3 {
    font-size: 4vw;
  }

  @media screen and (min-aspect-ratio: 2/3){
    flex-direction: column;
    height: 50vw;
    width: 17vw;
    .card{
      width: 22vw;
      margin: 1.5vw;
    }

    h2 {
      margin-top: 1vw;
      margin-bottom: 0;
      font-size: 2vw;
    }

    h3 {
      font-size: 3vw;
    }
  }

  @media screen and (min-aspect-ratio: 1/1){
    flex-direction: row;
    width: 100%;
    height: 8vw;
    margin: 0px;
    .card{
      height: 100%;
      width: 16vw;

      h2{
        font-size: 1.2vw;
        margin-bottom: 1vw;
        height: 1.5vw;
      }
      h3{
        font-size: 2.3vw;
        margin-bottom: 1vw;
      }
    }
  }
}
