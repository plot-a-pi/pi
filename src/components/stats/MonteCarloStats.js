import React from 'react';
import styles from './MonteCarloStats.css';
import PropTypes from 'prop-types';

const MonteCarloStats = ({ piApproximation, dartsTotal, circleTotal }) => {

  return (
    <>
      <div className={styles.MonteCarloStats}>
        <div className={styles.card}>
          <h2>Total Darts</h2>
          <h3>{dartsTotal}</h3>
        </div>
        <div className={styles.card}>
          <h2>Darts in Circle</h2>
          <h3>{circleTotal}</h3>
        </div>
        <div className={styles.card}>
          <h2>&pi; &asymp;</h2>
          <h3>{piApproximation.toFixed(4)}</h3>
        </div>
        <div className={styles.card}>
          <h2>% Error</h2>
          <h3>{(100 * Math.abs((Math.PI - piApproximation) / Math.PI)).toFixed(2)}</h3>
        </div>
      </div>
    </>
  );
};

MonteCarloStats.propTypes = {
  piApproximation: PropTypes.number.isRequired,
  dartsTotal: PropTypes.number.isRequired,
  circleTotal: PropTypes.number.isRequired
};

export default MonteCarloStats;
