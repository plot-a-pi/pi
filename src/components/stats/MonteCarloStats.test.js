import React from 'react';
import { shallow } from 'enzyme';
import MonteCarloStats from './MonteCarloStats';

describe('MonteCarloStats component', () => {
  it('renders the MonteCarloStats', () => {
    const wrapper = shallow(<MonteCarloStats piApproximation={3} dartsTotal={10} circleTotal={5} />);
    expect(wrapper).toMatchSnapshot();
  });
});
