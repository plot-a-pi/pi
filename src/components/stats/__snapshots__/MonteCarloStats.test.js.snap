// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonteCarloStats component renders the MonteCarloStats 1`] = `
<Fragment>
  <div
    className="MonteCarloStats"
  >
    <div
      className="card"
    >
      <h2>
        Total Darts
      </h2>
      <h3>
        10
      </h3>
    </div>
    <div
      className="card"
    >
      <h2>
        Darts in Circle
      </h2>
      <h3>
        5
      </h3>
    </div>
    <div
      className="card"
    >
      <h2>
        π ≈
      </h2>
      <h3>
        3.0000
      </h3>
    </div>
    <div
      className="card"
    >
      <h2>
        % Error
      </h2>
      <h3>
        4.51
      </h3>
    </div>
  </div>
</Fragment>
`;
